src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1884,25 +1884,29 @@ private void LogConflict(Reference reference, string fusionName, StringBuilder l
         /// <summary>
         /// Reads the state file (if present) into the cache.
         /// </summary>
-        private void ReadStateFile()
+        internal void ReadStateFile(FileExists fileExists)
         {
-            _cache = (SystemState)StateFileBase.DeserializeCache(_stateFile, Log, typeof(SystemState));
+            _cache = SystemState.DeserializeCacheByTranslator(_stateFile, Log);
 
             // Construct the cache if necessary.
             if (_cache == null)
```

Comment:
nit:
```suggestion
            if (_cache == null && AssemblyInformationCachePaths != null)
```
?

Calling `DeserializePrecomputedCachesByTranslator` if there are no precomputed caches looks a bit unexpected.